### PR TITLE
WIP: Add support for mem in the PBS headers (HPC-6209)

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -39,9 +39,12 @@
         # don't use installs from pbs-python from pypi
         # use local install from https://oss.trac.surfsara.nl/pbs_python/ticket/41#attachments
         # or rpms
-        'pbs_python >= 4.6',
+#        'pbs_python >= 4.6',
```
Comment (stdweird) [inline setup.py]: please undo

**Comment by itkovian** [comment]
Jenkins please test this

File: lib/vsc/jobs/pbs/submitfilter.py [hunk 2]
```diff
@@ -41,12 +41,13 @@
 PBS_OPTION_REGEXP = re.compile(r"(?:^|\s)-(\w+)(?:(?!\s*(?:\s-\w+|$))\s+(.*?(?=\s*(?:\s-\w+|$))))?")
 
 # TODO: all lower and uppercase combos?
-MEM_REGEXP = re.compile(r'^(p|v|pv)mem')
+MEM_REGEXP = re.compile(r'^(p|v|pv)+mem')
```
Comment (wpoely86) [inline lib/vsc/jobs/pbs/submitfilter.py]: don't you mean `?` instead of `+`?

Comment (itkovian) [inline lib/vsc/jobs/pbs/submitfilter.py]: Euh. Right.

File: lib/vsc/jobs/pbs/submitfilter.py [hunk 3]
```diff
@@ -41,12 +41,13 @@
 PBS_OPTION_REGEXP = re.compile(r"(?:^|\s)-(\w+)(?:(?!\s*(?:\s-\w+|$))\s+(.*?(?=\s*(?:\s-\w+|$))))?")
 
 # TODO: all lower and uppercase combos?
-MEM_REGEXP = re.compile(r'^(p|v|pv)mem')
+MEM_REGEXP = re.compile(r'^(p|v|pv)?mem')
```
Comment (JensTimmerman) [inline lib/vsc/jobs/pbs/submitfilter.py]: also make sure you have a test that fails when this regex is wrong

Comment (itkovian) [inline lib/vsc/jobs/pbs/submitfilter.py]: There really isn't a way here unless you have decent resource parsing. If the regex fails, the line can still be picked up as a general resource and thus make it to the new header.

**Comment by JensTimmerman** [comment]
you can write test input where you know it shouldn't end up in the output? so you check the output and see if it's not in there

**Comment by itkovian** [comment]
Sure you can. But this simply will not apply here. Unless we rewrite a big chunk of the 'parsing'.

File: lib/vsc/jobs/pbs/submitfilter.py (hunk 3 above)
Comment (stdweird) [inline lib/vsc/jobs/pbs/submitfilter.py]: you can unittest the regex itself

**Pull request merged**